hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/EntityRecordItemListener.java

```diff
@@ -380,4 +431,292 @@ private void updateEntity(
+            token.setTokenId(new Token.Id(EntityId.of(recordItem.getRecord().getReceipt().getTokenId())));
+
+            if (tokenCreateTransactionBody.hasFreezeKey()) {
+                token.setFreezeKey(tokenCreateTransactionBody.getFreezeKey().toByteArray());
+            }
+
+            if (tokenCreateTransactionBody.hasKycKey()) {
+                token.setKycKey(tokenCreateTransactionBody.getKycKey().toByteArray());
+            }
+
+            if (tokenCreateTransactionBody.hasSupplyKey()) {
+                token.setSupplyKey(tokenCreateTransactionBody.getSupplyKey().toByteArray());
+            }
+
+            if (tokenCreateTransactionBody.hasTreasury()) {
+                EntityId treasuryEntityId = EntityId.of(tokenCreateTransactionBody.getTreasury());
+                entityListener.onEntityId(treasuryEntityId);
+                token.setTreasuryAccountId(treasuryEntityId);
+            }
+
+            if (tokenCreateTransactionBody.hasWipeKey()) {
+                token.setWipeKey(tokenCreateTransactionBody.getWipeKey().toByteArray());
+            }
+
+            entityListener.onToken(token);
+        }
+    }
+
+    private void insertTokenDelete(RecordItem recordItem) {
+        if (entityProperties.getPersist().isTokens()) {
+            TokenDeleteTransactionBody tokenDeleteTransactionBody = recordItem.getTransactionBody().getTokenDeletion();
+            Optional<Token> optionalToken = retrieveToken(tokenDeleteTransactionBody.getToken());
+            if (optionalToken.isPresent()) {
+                Token token = optionalToken.get();
+                token.setModifiedTimestamp(recordItem.getConsensusTimestamp());
+                entityListener.onToken(token);
+            }
+        }
+    }
+
+    private void insertTokenDissociate(RecordItem recordItem) {
+        if (entityProperties.getPersist().isTokens()) {
+            TokenDissociateTransactionBody tokenDissociateTransactionBody = recordItem.getTransactionBody()
+                    .getTokenDissociate();
+            AccountID accountID = tokenDissociateTransactionBody.getAccount();
+            EntityId accountEntityId = EntityId.of(accountID);
+            entityListener.onEntityId(accountEntityId);
+
+            tokenDissociateTransactionBody.getTokensList().forEach(token -> {
+                insertTokenAccountUpdate(recordItem.getConsensusTimestamp(), token, accountID, true, false, false);
+            });
+        }
+    }
+
+    private void insertTokenAccountFreezeBody(RecordItem recordItem) {
+        if (entityProperties.getPersist().isTokens()) {
+            TokenFreezeAccountTransactionBody tokenFreezeAccountTransactionBody = recordItem.getTransactionBody()
+                    .getTokenFreeze();
+            AccountID accountID = tokenFreezeAccountTransactionBody.getAccount();
+            EntityId accountEntityId = EntityId.of(accountID);
+            entityListener.onEntityId(accountEntityId);
```

Comment:
Swap to do for Token since Account is covered by OnItem()
```suggestion
TokenID tokenID = tokenFreezeAccountTransactionBody.getToken();
EntityId.of(tokenID)
```